source/signaling_api.c

```diff
@@ -434,6 +434,171 @@ SignalingResult_t Signaling_ParseDescribeSignalingChannelResponse( const char *
+            {
+                result = SIGNALING_RESULT_UNEXPECTED_RESPONSE;
+            }
+            else
+            {
+                pCredentialsBuffer = pair.value;
+                credentialsBufferLength = pair.valueLength;
+            }
+        }
+        else
+        {
+            result = SIGNALING_RESULT_INVALID_JSON;
+        }
+    }
+
+    if( result == SIGNALING_RESULT_OK )
+    {
+        jsonResult = JSON_Iterate( pCredentialsBuffer, credentialsBufferLength, &( credentialsStart ), &( credentialsNext ), &( pair ) );
+
+        while( jsonResult == JSONSuccess )
+        {
+            if( strncmp( pair.key, "accessKeyId", pair.keyLength ) == 0 )
+            {
+                if( pair.valueLength < MAX_ACCESS_KEY_LEN )
+                {
+                    pCredentials->pAccessKeyId = pair.value;
+                    pCredentials->accessKeyIdLength = pair.valueLength;
+                }
+                else
+                {
+                    result = SIGNALING_RESULT_ACCESS_KEY_LENGTH_TOO_LARGE;
+                }
+                
+            }
+            else if( strncmp( pair.key, "secretAccessKey", pair.keyLength ) == 0 )
+            {
+                if( pair.valueLength < MAX_SECRET_KEY_LEN )
+                {
+                    pCredentials->pSecretAccessKey = pair.value;
+                    pCredentials->secretAccessKeyLength = pair.valueLength;
+                }
+                else
+                {
+                    result = SIGNALING_RESULT_SECRET_ACCESS_KEY_LENGTH_TOO_LARGE;
+                }
+            }
+            else if( strncmp( pair.key, "sessionToken", pair.keyLength ) == 0 )
+            {
+                if( pair.valueLength < MAX_SESSION_TOKEN_LEN )
+                {
+                    pCredentials->sessionToken = pair.value;
+                    pCredentials->sessionTokenLength = pair.valueLength;
+                }
+                else
+                {
+                    result = SIGNALING_RESULT_SESSION_TOKEN_LENGTH_TOO_LARGE;
+                }
+            }
+            else if( strncmp( pair.key, "expiration", pair.keyLength ) == 0 )
+            {
+                if( pair.valueLength < MAX_EXPIRATION_LEN )
+                {
+                    pCredentials->expiration = pair.value;
+                    pCredentials->expirationLength = pair.valueLength;
+                }
+                else
+                {
+                    result = SIGNALING_RESULT_EXPIRATION_LENGTH_TOO_LARGE;
+                }
+            }
+            else
+            {
+                /* Skip unknown attributes. */
+            }
+
+            if( result != SIGNALING_RESULT_OK )
+            {
+                break;
+            }
+
+            jsonResult = JSON_Iterate( pCredentialsBuffer, credentialsBufferLength, &( credentialsStart ), &( credentialsNext ), &( pair ) );
+        }
+    }
+

```

Comment:
Should we ensure that accessKeyId/secretAccessKey and sessionToken are all parsed in this message?